manifests/init.pp

```diff
@@ -46,6 +47,7 @@
   Stdlib::Absolutepath $rubypath,
   Boolean $manage_bin_symlinks = false,
   Integer $facts_refresh_interval,
+  String[1] $periodic_runmode,
```

Comment:
As mentioned, a stricter data type is welcome. Also the default values being the same for all OS, putting it here instead of in Hiera makes it possible for puppet-strings to display the actual default version.

```suggestion
  Enum['cron', 'systemd.timer'] $periodic_runmode = 'cron',
```